### PR TITLE
conversion of WfBinaryBufferTest.java tests

### PR DESCRIPTION
**Pull request by PhillyWebGuy**

### CODE REVIEW

File: src/wf_buffer/tests.rs [hunk 1]
```diff
@@ -119,6 +120,24 @@ fn test_append_bits_1() {
...
+
+    begin = concatinate_bits(&begin, 0, &byte_array_1, 24);
+
+    assert_eq!(begin.len() * BYTE, 24, "Binary buffer length should be 24 bits");
+    assert_eq!(&byte_array_2.len() * BYTE, 16, "Binary buffer length should be 16 bits");
+    assert_eq!(to_hex(&begin), "e63887", "Byte array 1 should have been correctly added to the binary buffer");
+
+    let end = concatinate_bits(&begin, 24, &byte_array_2, 12);
+    
+    assert_eq!(end.len() * BYTE, 36, "Binary buffer length should be 36 bits");
```
Comment (PhillyWebGuy) [inline src/wf_buffer/tests.rs:138]: @isaacadams This actually returns 40. Is it possible concatinate_bits has a bug or am I missing something super obvious?

Comment (PhillyWebGuy) [reply src/wf_buffer/tests.rs:138]: In theory, that 4th argument should preclude that from happening.

File: src/wf_buffer/tests.rs [hunk 2]
```diff
@@ -119,6 +120,51 @@ fn test_append_bits_1() {
...
+
+    let byte_array_1: Vec<u8> = vec![0xDD, 0xFF]; // 1101 1101 | 1111 1111
+    let byte_array_2: Vec<u8> = vec![0xBF]; // 1011 1111
+    let mut begin: Vec<u8> = vec![];
+
+    assert_eq!(begin.len() * BYTE, 0, "Binary buffer length should be 0 bits");
+    assert_eq!(byte_array_1.len() * BYTE, 16, "byte_array_2 length should be 16 bits");
+    
+    let end = concatinate_bits(&begin, 0, &byte_array_1, 4);
+    assert_eq!(end.len() * BYTE, 4, "Binary buffer length should be 4 bits");
```
Comment (PhillyWebGuy) [inline src/wf_buffer/tests.rs:152]: @isaacadams This also fails.

File: src/wf_buffer/tests.rs [hunk 3]
```diff
@@ -119,6 +125,182 @@ fn test_append_bits_1() {
...
+        result,
+        extract_bits(&byte_array_1, 32, 12, 16),
+        "Should have correctly extracted 10 bits from binary buffer"
+    );
+}
+
+#[test]
+fn test_add_field_utf() {
+
+    let mut buffer: Vec<Field> = vec![];
```
Comment (PhillyWebGuy) [inline src/wf_buffer/tests.rs:278]: Do we want <Field> or do we need to encode first and store as <u8>?

Comment (PhillyWebGuy) [reply src/wf_buffer/tests.rs]: @isaacadams Hey, dumb question, can you point me to which method that is you ported?

Comment (isaacadams) [reply src/wf_buffer/tests.rs]: fair question: https://github.com/fennelLabs/whiteflag-rust/pull/14

File: src/wf_field/field.rs [hunk 4]
```diff
@@ -115,7 +115,7 @@ impl Field {
         message_buffer: &[u8],
         message_buffer_bit_length: usize,
         start_bit: usize,
-    ) -> usize {
+    ) -> String {
```
Comment (PhillyWebGuy) [inline src/wf_field/field.rs:118]: @isaacadams I started to change this... I'm thinking your intended functionality here is different from what ended up in the code base? At the very least, the comments above the function don't match what it does.

Comment (isaacadams) [reply src/wf_field/field.rs:94]: The comments were straight copied from the java codebase merely for the sake of being easy to reference, while some code of each function was slightly adjusted to get things working within the rust paradigm. Therefore, yes, some functions may return different values than what the java codebase comments says it does.

One of the goals for this milestone is to get the comments fixed up.

Also, good call with getting this method fixed up. The `Field` struct has been significantly changed in my changes from a few days ago. However, I did make related adjustments according to your suggestion here: https://github.com/fennelLabs/whiteflag-rust/pull/22/commits/0a85972370d3a4798e70a230448553c694309b61

**Review by Romulus10 — APPROVED** [review]

**Pull request merged**